src/tunnel.cc

```diff
@@ -947,8 +936,8 @@ tunnelConnectedWriteDone(const Comm::ConnectionPointer &conn, char *, size_t len
     tunnelState->client.writer = NULL;
 
     if (flag != Comm::OK) {
-        *tunnelState->status_ptr = Http::scInternalServerError;
-        tunnelErrorComplete(conn->fd, data, 0);
+        auto err = new ErrorState(ERR_WRITE_ERROR, Http::scInternalServerError, tunnelState->request.getRaw(), tunnelState->al);
```

Comment:
Please detail this error by passing the saved `errno` (currently the anonymous parameter; see `xerrno` in `IOCB`) to the ErrorState object.

```suggestion
        const auto err = new ErrorState(ERR_WRITE_ERROR, Http::scInternalServerError, tunnelState->request.getRaw(), tunnelState->al);
        err->xerrno = xerrno;
```

Reply:
Done.